app/src/services/market_maker.ts

```diff
@@ -61,30 +62,17 @@ class MarketMakerService {
     return this.contract.removeFunding(amount)
   }
 
-  /* TODO: TBD */
   getActualPrice = async (): Promise<{ actualPriceForYes: number; actualPriceForNo: number }> => {
-    // let [actualPriceForYes, actualPriceForNo] = await Promise.all([
-    //   this.contract.calcMarginalPrice(0),
-    //   this.contract.calcMarginalPrice(1),
-    // ])
-    //
-    // const two = new BigNumber(2)
-    // const twoPower64 = two.pow(64)
-    //
-    // actualPriceForYes =
-    //   actualPriceForYes
-    //     .mul(new BigNumber(10000))
-    //     .div(twoPower64)
-    //     .toNumber() / 10000
-    // actualPriceForNo =
-    //   actualPriceForNo
-    //     .mul(new BigNumber(10000))
-    //     .div(twoPower64)
-    //     .toNumber() / 10000
-    //
+    const { address } = this.contract
+    const { balanceOfForYes, balanceOfForNo } = await this.getBalanceInformation(address)
```

Comment:
Not sure about this. We are fetching the balance information twice for no reason, and it worries me a little that we are already doing a lot of concurrent requests to the contract here. I think we should use the result of the balances in the `useMarketMakerData` hook to compute the prices.

Reply:
Ok, I use the result of the balances in the useMarketMakerData, and pass the value as a parameter to compute the prices.